components/prism-core.js

```diff
@@ -181,7 +247,41 @@ var _ = {
 		}
 	},
 
+	/**
+	 * This namespace contains all currently loaded languages and the some helper functions to create and modify languages.
+	 *
+	 * @namespace
+	 * @memberof Prism
+	 * @public
+	 */
 	languages: {
+		/**
+		 * Creates a deep copy of the language with the given id and appends the given tokens.
+		 *
+		 * If a token in `redef` also appears in the copied language, then the existing token in the copied language
+		 * will be overwritten at its original position.
+		 *
+		 * ## Best practices
+		 *
+		 * Since the position of overwriting tokens (token in `redef` that overwrite tokens in the copied language)
+		 * doesn't matter, they can technically be in any order. However, this can be confusing to others that trying to
+		 * understand the language definition because, normally, the order of tokens matters in Prism grammars.
+		 *
+		 * Therefore, it is encouraged to order overwriting tokens according to the positions of the overwritten tokens.
+		 * Furthermore, all non-overwriting tokens should be placed after the overwriting ones.
+		 *
+		 * @param {string} id The id of the language to extend. This has to be a key in `Prism.languages`.
+		 * @param {Grammar} redef The new tokens to append.
+		 * @returns {Grammar} The new language created.
+		 * @public
+		 * @example
+		 * Prism.languages['css-with-colors'] = Prism.languages.extend('css', {
+		 *     // this will overwrite the 'comment' token of CSS at its original position
+		 *     'comment': { ... },
+		 *     // this will be appended
+		 *     'color': /\b(?:red|green|blue)\b/
```

Comment:
I think these examples need some explanation in the prose. Why is `color` appended & `comment` overridden?

Reply:
I'll add something.

On that note: ["override" or "overwrite"](https://stackoverflow.com/a/8651622/7595472)?

Reply:
"overwrite", I think. We're fully replacing the object, yeah?

Reply:
We do. Then I'll stick with "overwrite".